File: DesignPatterns/src/Java/002-GOF-Patterns/002-StructuralPatterns/001-Composite/EmployeeApp/src/org/csystem/app/Worker.java
package org.csystem.app;

public class Worker extends Employee {
    private int m_hourPerDay;
    private double m_feePerHour;

    public Worker(String citizenId, String title, String name, String address, int hourPerDay, double feePerHour)
    {
        super(citizenId, title, name, address);
        m_hourPerDay = hourPerDay;
        m_feePerHour = feePerHour;
    }

    public int getHourPerDay()
    {
        return m_hourPerDay;
    }

    public void setHourPerDay(int hourPerDay)
    {
        m_hourPerDay = hourPerDay;
    }

    public double getFeePerHour()
    {
        return m_feePerHour;
    }

    public void setFeePerHour(double feePerHour)
    {
        m_feePerHour = feePerHour;
    }

    @Override
    public double calculateInsurance()
    {
        return m_feePerHour * m_hourPerDay * 30;
    }
}

File: DesignPatterns/src/C++/001-SOLID/003-OCP/OCPInheritance/Worker.cpp
#include <iostream>
#include "Worker.hpp"

using namespace std;

Worker::Worker(const char *name, int no, Shift shift, int hourPerDay, double feePerHour)
    : Employee{ name, no }, m_shift {shift}, m_feePerHour{feePerHour}, m_hourPerDay {hourPerDay}
{}

void Worker::disp() const
{
	const char *shifts[]{ "Morning", "Noon", "Evenning" };

	Employee::disp();
	cout << shifts[m_shift] << endl;
}

double Worker::GetInsurancePayment() const
{
    return m_feePerHour * m_hourPerDay * 30;
}
